src/hazelcore/Statics_Typ.re

```diff
@@ -0,0 +1,19 @@
+let rec syn = (ctx: Contexts.t, ty: HTyp.t): Kind.t =>
+  switch (ty) {
+  | TyVar(idx, _) =>
+    let (_, k) = List.nth(ctx.tyvars, idx);
+    k;
+  | TyVarHole(_, _)
+  | Hole => KHole
+  | Int
+  | Float
+  | Bool
+  | Prod([]) => Type
+  | Arrow(ty1, ty2)
+  | Sum(ty1, ty2) =>
+    ana(ctx, ty1, Kind.Type) && ana(ctx, ty2, Kind.Type) ? Type : KHole
```

Comment:
if analysis fails on either argument, then synthesis should fail, not return KHole. need to change return type to option.